--- a/telecom/recoverymgrd/conf_lex.l
+++ b/telecom/recoverymgrd/conf_lex.l
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 #include <string.h>
 
+#define  YY_NO_UNPUT
 #define  YYSTYPE char *
 #include "conf_yacc.h"
 extern int yylex(void);
@@ -42,42 +43,6 @@
 
 %}
 
-%{
-int yyget_lineno(void);
-%}
-%{
-FILE *yyget_in(void);
-%}
-%{
-FILE *yyget_out(void);
-%}
-%{
-int yyget_leng(void);
-%}
-%{
-char *yyget_text(void);
-%}
-%{
-void yyset_lineno(int);
-%}
-%{
-void yyset_in(FILE *);
-%}
-%{
-void yyset_out(FILE *);
-%}
-%{
-int yyget_debug(void);
-%}
-%{
-void yyset_debug(int);
-%}
-%{
-int yylex_destroy(void);
-%}
-
-%option nounput
-
 %%
 
 PID			{return PID;};
