From: Simon Horman <horms@verge.net.au>
To: Petter Reinholdtsen <pere@hungry.com>, 459497@bugs.debian.org
Cc: linux-ha-dev@lists.linux-ha.org
Subject: [patch] init: no need to start ntp as $time is a required-start

See http://wiki.debian.org/LSBInitScripts

Cc: Petter Reinholdtsen <pere@hungry.com>
Signed-off-by: Simon Horman <horms@verge.net.au>

Index: heartbeat/heartbeat/init.d/heartbeat.in
===================================================================
--- heartbeat.orig/heartbeat/init.d/heartbeat.in	2008-01-08 11:56:04.000000000 +0900
+++ heartbeat/heartbeat/init.d/heartbeat.in	2008-01-08 11:57:15.000000000 +0900
@@ -37,7 +37,7 @@
 # Short-Description: High-availability services.
 # Provides: heartbeat HA
 # Required-Start: $remote_fs $network $time $syslog
-# Should-Start: ntp openhpid
+# Should-Start: openhpid
 # Required-Stop: $remote_fs $network $time $syslog
 # Should-stop: openhpid
 # Default-Start: 2 3 4 5
